### ISSUE TITLE
Readdress Overall Purpose

### ISSUE THREAD

**Issue by Apexal** [Apexal/late#19]
### What are we building?

The original proposal called for basically an extension to Google Calendar. We have already established we are not going to limit ourselves to Google Calendar and instead are going for a less centralized approach. During our meetings we have often touched on things originally considered outside the scope of our project.

I've been thinking (I'm in a Data Structures lecture and not paying attention) and perhaps we need to reevaluate our project scope and determine the long-term goals of LATE.

Are we creating just an semi-automated scheduling assistant for assignments only? Or are perhaps building a more encompassing service that could handle more than just allocating your free time?

We could build this scheduling assistant part as the main goal of our semester now, but with the overall goal in mind of the schedule part just being one of potentially many components of LATE.

Imagine at some future point in time, a student logs into LATE on their phone and sees:
- a quick overview of what class they should currently be in (where it is and when) along with assignments due that class (and whether or not they were completed)
- a reminder of where and when next class is again along with reminders of associated work
- an overview of how today's free time should be spent working/studying
- a list of upcoming assignments, exams, etc.

There are many possible uses that can be added once schedule management is created.

Perhaps in future semesters I will continue this project for anyone current or new that wants to further and further expand it.

What do you guys think?

**Comment by DrSeaTurtle** [comment]
LATE could stand to improve by borrowing concepts from task managing programs: things like recurring events,  support for dynamic schedules(so it doesn't just become meaningless if your lecture is canceled or something), sub-tasks, etc.  None of these are particularly necessary or will be something I'd feel would be a failure of LATE if we turned up in December without these, but personaly LATE would occupy a place closer to something like Todoist rather than google calendar. That's just my 2 cents though.

TL;DR task manager-esque functionalities would benefit LATE but aren't really needed

**Comment by kvverti** [comment]
My thought on Late is that it allocates time for middle-term or long-term asks. Anything that can be split easily into parts and has a due date in the semi near future. They are not necessarily limited to assignments. The selling point is that Late will automatically allocate time for these fuzzy middle-term tasks around your existing schedule. It would particularly help in divvying time among multiple assignments, which can otherwise be confusing to do.

**Comment by AbigailMedina** [comment]
My thoughts on LATE's development for this semester:
Core Purpose:
Replace the need for lazy students to figure out what they should be doing at any given moment.

How we are going to do it:
	1)Automatically schedule work times for specific tasks around daily schedule.
	2)Remind student when they should be doing a specific task
	3)Update scheduled work times based on their completion of task

Use Case 1:

Student gets hw assignment from class A. They open LATE and see VIEW 1. 
They navigate to VIEW 2, which lets them add the assignment with appropriate information associated. While in VIEW 2, they also remove the last assignment from class A that they just finished.
The student navigates to VIEW 1 to see their current work lineup and leaves LATE.

Use Case 2:

Student gets text from LATE saying that they are scheduled to be studying for class A test in 30 mins for two hours. They dismiss the text and walk to the library to study. :) When the two hours is up, student gets a text from LATE saying its time to go have fun and to reply y if they had spent the allotted time studying. Student responds y and closes phone.

**Comment by wangbowen1401** [comment]
Apologies in advance if any of the comment is incorrect, but since I still didn't meet with the team, here is my perspective of the discussion. 

Late should focus on simplifying the calendar making process, or else people are just better off using google calendar, so just construct it as a more student-focused google calendar. 

Late will have a large margin of error during early launch, just because the program doesn't have enough information regarding an individual user. Studying for 2 hours might not be sufficient for every student, but determining the average time from a survey is a good start. And yes, it should become more accurate over time, but that will be data analyst stuff. (Maybe an average time correction factor for an individual user which accumulates over time and we just multiply that value by the average time of all the students?) 

For now, probably focus on generating time commitments with the average time required, if at the end of a specific task, the user is not satisfied with the progress or he/she has completed the task before the delegated time, the program should automatically update the rest of the week/month accordingly. (Might be too many interactions required though)

Interesting Function: Add a files reader which automatically generates the academic schedule based on course syllabus?

**Comment by Apexal** [comment]
@wangbowen1401 Here is the short summary of the proposal and slight changes we have made after discussing:
- **LATE** is a web app that, given your academic schedule, takes your assignments and time estimates for each assignment as *input* and as *output* tells you when you should work on each item and for how long by distributing the work throughout the free time in your schedule.

Originally the entire focus was on integrating with Google Calendar because that is the tool I personally use to manage all my schedules and free time. However, once the team was formed and we found not everyone uses Google Calendar as heavily, we decided we should not so tightly couple ourselves to Google Calendar but offer options for users to input their schedules.

We still need to determine exactly what this means.
